File: 기본 수학/1002/1002.cpp
#include <iostream>
#define dist(x1, y1, x2, y2) (x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2)
using namespace std;

int main()
{
   int n, x1, y1, r1, x2, y2, r2, d, answer;
   cin>>n;

   for (int i = 0; i < n; ++i)
   {
      cin>>x1>>y1>>r1>>x2>>y2>>r2;
      d = dist(x1, y1, x2, y2);
      if (x1 == x2 and y1 == y2 and r1 == r2)
         answer = -1;
      else if (d == (r1 + r2) * (r1 + r2) or d == (r1 - r2) * (r1 - r2))
         answer = 1;
      else if ((r1 - r2) * (r1 - r2) < d and d < (r1 + r2) * (r1 + r2))
         answer = 2;
      else
         answer = 0;
      cout<<answer<<"\n";
   }

   return 0;
}

File: 기본 수학/1002/1002.py
def dist(x1, y1, x2, y2):
   return math.sqrt((x1 - x2) ** 2 + (y1 - y2) ** 2)

n = int(input())

for i in range(n):
   x1, y1, r1, x2, y2, r2= map(int, input().split())
   d = dist(x1, y1, x2, y2)
   if (x1 == x2 and y1 == y2 and r1 == r2):
      answer = -1
   elif (d == r1 + r2 or d == abs(r1 - r2)):
      answer = 1
   elif (abs(r1 - r2) < d < r1 + r2):
      answer = 2
   else:
      answer = 0
   print(answer)